radio/src/targets/common/arm/stm32/audio_dac_driver.cpp

```diff
@@ -192,38 +219,42 @@ void audioInit()
 
 void audioEnd()
 {
-  DAC->CR = 0 ;
-  AUDIO_TIMER->CR1 = 0 ;
-  // Also need to turn off any possible interrupts
-  NVIC_DisableIRQ(AUDIO_TIM_IRQn) ;
-  NVIC_DisableIRQ(AUDIO_DMA_Stream_IRQn) ;
-}
+  DAC->CR = 0;
+  AUDIO_TIMER->CR1 = 0;
 
-extern "C" void AUDIO_TIM_IRQHandler()
-{
-  DEBUG_INTERRUPT(INT_AUDIO);
-  DAC->CR &= ~DAC_CR_DMAEN1 ;     // Stop DMA requests
-#if defined(STM32F2)
-  DAC->CR &= ~DAC_CR_DMAUDRIE1 ;  // Stop underrun interrupt
-#endif
-  DAC->SR = DAC_SR_DMAUDR1 ;      // Write 1 to clear flag
+  // Also need to turn off any possible interrupts
+  NVIC_DisableIRQ(AUDIO_TIM_IRQn);
+  NVIC_DisableIRQ(AUDIO_DMA_Stream_IRQn);
 }
 
 extern "C" void AUDIO_DMA_Stream_IRQHandler()
 {
-  AUDIO_DMA_Stream->CR &= ~DMA_SxCR_TCIE ;            // Stop interrupt
-  AUDIO_DMA->HIFCR = DMA_HIFCR_CTCIF5 | DMA_HIFCR_CHTIF5 | DMA_HIFCR_CTEIF5 | DMA_HIFCR_CDMEIF5 | DMA_HIFCR_CFEIF5 ; // Write ones to clear flags
-  AUDIO_DMA_Stream->CR &= ~DMA_SxCR_EN ;                              // Disable DMA channel
+  // disable transfer complete interrupt
+  AUDIO_DMA_Stream->CR &= ~DMA_SxCR_TCIE;
 
-  if (nextBuffer) audioQueue.buffersFifo.freeNextFilledBuffer();
+  // reset DMA flags
+  AUDIO_DMA->HIFCR = DMA_HIFCR_CTCIF5 | DMA_HIFCR_CHTIF5 | DMA_HIFCR_CTEIF5 |
+                     DMA_HIFCR_CDMEIF5 | DMA_HIFCR_CFEIF5;
 
+  // Disable DMA stream
+  AUDIO_DMA_Stream->CR &= ~DMA_SxCR_EN;
```

Comment:
aren't there inline helper functions for this stuff in the CMSIS device headers?

Reply:
this will be done later